src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -395,7 +396,10 @@ private function createDelayQueue(int $delay, ?string $routingKey, bool $isRetry
             // after being released from to DLX, make sure the original routing key will be used
             // we must use an empty string instead of null for the argument to be picked up
             'x-dead-letter-routing-key' => $routingKey ?? '',
-        ]);
+        ];
+
+        $delayArguments = $this->connectionOptions['delay']['arguments'] ?? [];
+        $queue->setArguments(array_merge($defaultArguments, $delayArguments));
```

Comment:
Just a idea, to have not 2 vars only used once, but just a hint from my side ...
```suggestion
        $queue->setArguments(array_merge([
            'x-message-ttl' => $delay,
            // delete the delay queue 10 seconds after the message expires
            // publishing another message redeclares the queue which renews the lease
            'x-expires' => $delay + 10000,
            // message should be broadcasted to all consumers during delay, but to only one queue during retry
            // empty name is default direct exchange
            'x-dead-letter-exchange' => $isRetryAttempt ? '' : $this->exchangeOptions['name'],
            // after being released from to DLX, make sure the original routing key will be used
            // we must use an empty string instead of null for the argument to be picked up
            'x-dead-letter-routing-key' => $routingKey ?? '',
        ], $this->connectionOptions['delay']['arguments'] ?? []));
```